### ISSUE TITLE
Bug - SHOW_ALL scale mode doesn't show all

### ISSUE THREAD

**Issue by hdodov** [photonstorm/phaser#2732]
I looked in the docs for a configurable that I must set for this to work, but I couldn't find it, so I'm guessing it's a bug.

In the create function, I have:
```
game.scale.scaleMode = Phaser.ScaleManager.SHOW_ALL;
```
I also have this:
```
window.onresize = function(){
	var w = window.innerWidth, h = window.innerHeight;
	if(w < 400 || h < 500){ w = 400; h = 500; }
	game.scale.setGameSize(w, h);
};
```

What I want is my game to be whatever size the screen is, if its width is greater than 400 and its height is greater than 500. Else, I want the game to be exactly 400x500 and to be scaled appropriately so that it's entirely visible on the screen.

If the screen is 200x500, the scale mode works fine. The whole game is visible and is scaled to 0.5 of its original scale because I've set the game width to 400 and it's 200. This leaves a 250 pixel empty space below the game (which is expected) because its scaled in half, which means that half of the screen is empty.

However, if the screen is 400x250, the scale mode works weirdly. This time, the game is not entirely visible and its scale is 1, instead of 0.5. Instead of the screen being half empty (this time on the side, not on the bottom), the game is half hidden.

The engine sets the game's width to fit entirely in the screen's width, but doesn't care about the height. I.E. if the width of the game is too big, it scales it appropriately, but if the height is too big, it doesn't scale it down and just sticks to setting the width to the maximum.

TL;DR if width is smaller than height, scale mode works fine. If it's the other way around, it's like the scale mode does nothing.

**The image below shows it best.**

![img](https://cloud.githubusercontent.com/assets/5570098/18126865/d6ee5cd8-6f85-11e6-93bb-74ffdaa40653.jpg)

**Comment by samme** [comment]
Hi hdov, does the game scale as you expect when you remove your `window.onresize` handler?

**Comment by hdodov** [comment]
@samme yep.

**Comment by samme** [comment]
Hm, in that case the problem may not be SHOW_ALL.

Try using http://phaser.io/docs/2.6.2/Phaser.ScaleManager.html#onSizeChange instead of `window.resize`.

**Comment by hdodov** [comment]
Oops, excuse me. I thought you wrote "does my game have the same problem when I remove the handler". It doesn't scale right. Without the `window.onresize`, the problem is still there. On my PC screen, it's reeeally scaled up and more than half of it is hidden (just like in the image I've shown).

**Comment by samme** [comment]
OK, I made a pen: http://codepen.io/anon/pen/wzBwKP . SHOW_ALL seems to work correctly there.

**Comment by hdodov** [comment]
Hmmm, very strange. Turns out the problem was somewhere in the definition of my game. When I did it like this
```
var game = new Phaser.Game(
	"100", // width
	"100", // height
	Phaser.AUTO,
	"game-canvas-holder",
	{
		preload:preload,
		create:create
	},
	true, // transparent background
	true // anti-aliasing
);
```
But when I did it like you:
```
var game = new Phaser.Game({
	scaleMode: Phaser.ScaleManager.SHOW_ALL,
	state: {
		preload:preload,
		create:create
	}
});
```
It started working. Even with my resize handler. Well, turns out it's not a bug after all, excuse me. Can I know what exactly was causing the problem, though?

**Comment by hdodov** [comment]
The problem was in my game's parent div. When I removed it so that my game is directly in the body, it started working. It's strange, though, that container didn't have any CSS styling it.

**Comment by medicmilos** [comment]
Hi everyone, i have one interesting problem with Phaser scaling (I think). I forked @samme pen: [https://codepen.io/medicmilos/pen/dBooWK](https://codepen.io/medicmilos/pen/dBooWK) and added debug pointer, and as you can see if you move your mouse towards bottom right of the image you will see that near the end pointer disappears. Is that known problem, or am I doing something wrong?

**Comment by photonstorm** [comment]
@medicmilos this issue was closed 3 years ago, posting a question here isn't going to be seen by anyone I'm afraid. I would strongly recommend using the forum instead.
